docs/src/main/sphinx/security/jwt-auth.rst

```diff
@@ -0,0 +1,158 @@
+methods:
+
+.. code-block:: properties
+
+    http-server.authentication.type=PASSWORD,JWT
+    http-server.authentication.jwt.key-file=https://cluster.example.net/.well-known/jwks.json
+
+The following configuration properties are available:
+
+.. list-table:: Configuration properties for JWT authentication
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - ``http-server.authentication.jwt.key-file``
+     - Required. Specifies either the URL to a JWKS service or the path to a
+       PEM or HMAC file, as described below this table.
+   * - ``http-server.authentication.jwt.required-issuer``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``iss`` field to consider this JWT valid for this Trino server.
+       The ``iss`` field in the JWT identifies the principal that issued the
+       JWT.
+   * - ``http-server.authentication.jwt.required-audience``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``aud`` field to consider this JWT valid for this Trino server.
+       The ``aud`` field in the JWT identifies the recipients that the JWT is
+       intended for.
+   * - ``http-server.authentication.jwt.principal-field``
+     - Optional. String to identify the field in the JWT that identifies the
+       subject of the JWT. The default value is ``sub``.
+   * - ``http-server.authentication.jwt.user-mapping.pattern``
+     - Optional. A regular expression pattern to :doc:`map all user names
+       </security/user-mapping>` for this authentication system to the format
+       expected by the Trino server.
+   * - ``http-server.authentication.jwt.user-mapping.file``
+     - Optional. The path to a JSON file that contains a set of
+       :doc:`user mapping rules </security/user-mapping>` for this
+       authentication system.
+
+Use the ``http-server.authentication.jwt.key-file`` property to specify
+either:
+
+-  The URL to a JWKS endpoint service, where the URL begins with ``https://``
+   and ends with the port number. The JWKS service must be reachable from the
+   coordinator. If the coordinator's local network locks down port access for
+   security, then administrators must either open a port to the JWKS, or look in
+   the server's ``/.well-known/jwks.json`` file. `RFC 5785
+   <https://tools.ietf.org/html/rfc5785>`_ describes the ``.well-known``
+   mechanism.
+
+   .. caution::
+
+        The Trino server also accepts JWKS URLs that begin with ``http://``, but
+        using this protocol results in a severe security risk. Only use this
+        protocol for short-term testing during development of your cluster.
+
+-  The path to a local file in :doc:`PEM </security/inspect-pem>` or `HMAC
+   <https://en.wikipedia.org/wiki/HMAC>`_ format that contains one or more
+   certificates that validate the Trino coordinator.
+
+   If the file contains more than one certificate, you can use ``$KEYID`` as
```

Comment:
This is a bit confusing... the path points to a file that contains a single key (no certificates here).  If the file path contains `$KEYID` then we interpolate the keyid from the JWT into the file path before loading this key.  They enables support for setups with multiple keys.

Reply:
Used your wording.